client/src/components/Carousel.jsx

```diff
@@ -0,0 +1,62 @@
+import { useState, useEffect } from "react";
+import { RxDot, RxDotFilled } from 'react-icons/rx';
+import { HiOutlineChevronLeft, HiOutlineChevronRight } from 'react-icons/hi'
+
+export default function Carousel({ images }) {
+  const [current, setCurrent] = useState(0);
+
+  useEffect(() => {
+    const intervalId = setInterval(handleRightButton, 5000);
+    return () => clearInterval(intervalId);
+  });
+
+  function handleLeftButton() {
+    setCurrent(((current - 1) + images.length) % images.length)
+  }
+
+  function handleRightButton() {
+    setCurrent((current + 1) % images.length)
+  }
+
+  return (
+    <div>
+      <div className="top-row">
+        <LeftButton click={handleLeftButton} />
+        <img className={current === 4 ? "spin" : undefined} alt={images[current].name} src={images[current].src} />
```

Comment:
You might want to add a "spinnable" field to `images` that you can use to decide whether or not to set the spin class. That would make your code more data-driven.